pkg/eventprocessor/impl/executor/impl/blockscope.go

```diff
@@ -162,6 +165,54 @@ func (bs *blockScope) TxnReceiptExists(ctx context.Context, txnHash common.Hash)
 	return true, nil
 }
 
+func (bs *blockScope) StateHash(ctx context.Context, chainID tableland.ChainID) (executor.StateHash, error) {
+	hash, err := dbhash.DatabaseStateHash(ctx, bs.txn, []dbhash.Option{
+		dbhash.WithFetchSchemasQuery(
+			fmt.Sprintf(`SELECT tbl_name, sql 
+				FROM sqlite_schema
+			    WHERE name NOT LIKE 'sqlite_%%'  
+				AND name LIKE '%%\_%d\_%%' ESCAPE '\'
+				AND type = 'table'
+				UNION ALL
+				SELECT tbl_name, sql 
+				FROM sqlite_schema
+				WHERE name in ('registry', 'system_acl', 'system_controller', 'system_txn_receipts')
+				ORDER BY tbl_name COLLATE nocase;`, chainID),
+		),
+		dbhash.WithPerTableQueryFn(func(tableName string) string {
```

Comment:
This is a bit ugly but each system table requires a particular query to fetch its data. That's because we have to exclude non-deterministic columns such as `created_at` and add specific `ORDER BY` clauses.